src/Products/MeetingCommunes/browser/overrides.py

```diff
@@ -21,21 +20,65 @@
 from Products.PloneMeeting.utils import get_person_from_userid
 from zope.component import getAdapter
 
+import cgi
+
 
 class MCItemDocumentGenerationHelperView(ItemDocumentGenerationHelperView):
     """Specific printing methods used for item."""
 
-    def print_all_annexes(self, portal_types=['annex']):
-        ''' Printing Method use in templates :
-            return all viewable annexes for item '''
+    def print_all_annexes(self, portal_types=['annex'], filters={}, with_icon=False, long_format=False):
+        """
+        Printing Method use in templates :
+        return all viewable annexes for item
+        @param: filters is a dict of {"attribute" : value}.
+        It excludes all non matching annex from the result.
+        Example : {'confidential': True, 'publishable': False}
+        possible keys are : 'confidential', 'to_print', 'to_sign' and 'publishable' (all are bool or None)
+        """
         res = []
         annexes = get_annexes(self.context, portal_types=portal_types)
+        mimetypes_registry = self.portal.mimetypes_registry
+        if filters:
+            effective_annexes = []
+            for annex in annexes:
+                use_this_annex = True
+                for attribute in filters:
```

Comment:
@odelaere je préfère la notation "for attribute, value in filters.items(): ..." et réutiliser "value" au lieu de ré-accéder le dict via "filter[attribute]" plus lisible à mon avis...